src/naive/date.rs

```diff
@@ -992,19 +1012,20 @@ impl NaiveDate {
     /// ```
     #[inline]
     #[must_use]
-    pub fn and_hms_nano_opt(
+    pub const fn and_hms_nano_opt(
         &self,
         hour: u32,
         min: u32,
         sec: u32,
         nano: u32,
     ) -> Option<NaiveDateTime> {
-        NaiveTime::from_hms_nano_opt(hour, min, sec, nano).map(|time| self.and_time(time))
+        let time = try_opt!(NaiveTime::from_hms_nano_opt(hour, min, sec, nano));
```

Comment:
Can we just `?` instead of using `try_opt!()`?

Reply:
`?` is not available in const, that is the reason to have this macro.

Reply:
Ok.